### PR TITLE
Using Opencv3.0.0

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

File: cv_bridge/src/module.cpp [hunk 1]
```diff
@@ -227,16 +229,16 @@ int convert_to_CvMat2(const PyObject* o, cv::Mat& m)
 
     if( m.data )
     {
-        m.refcount = refcountFromPyObject(o);
+        //m.refcount = refcountFromPyObject(o);
```
Comment (vrabaud) [inline cv_bridge/src/module.cpp:232]: why is that commented out ?

File: cv_bridge/src/module.cpp [hunk 2]
```diff
@@ -145,7 +147,7 @@ void NumpyAllocator::deallocate( int* refcount, uchar* datastart, uchar* data )
 }
 
 // Declare the object
-NumpyAllocator g_numpyAllocator;
+NumpyAllocator *g_numpyAllocator;
```
Comment (vrabaud) [inline cv_bridge/src/module.cpp:150]: this is just for speed right ?

File: cv_bridge/src/module.cpp [hunk 3]
```diff
@@ -155,7 +157,7 @@ int convert_to_CvMat2(const PyObject* o, cv::Mat& m)
     if(!o || o == Py_None)
     {
         if( !m.data )
-            m.allocator = &g_numpyAllocator;
+            m.allocator = g_numpyAllocator;
```
Comment (vrabaud) [inline cv_bridge/src/module.cpp:160]: where is the pointer g_numpyAllocator being initialized ?

File: cv_bridge/src/module.cpp (hunk 1 above)
Comment (edgarriba) [inline cv_bridge/src/module.cpp:232]: in the new Mat interface doesn't exist refcount attribute anymore

File: cv_bridge/src/module.cpp (hunk 2 above)
Comment (edgarriba) [inline cv_bridge/src/module.cpp:150]: i think it's also to satisfy m.allocator

//! custom allocator
MatAllocator\* allocator;
//! and the standard allocator
static MatAllocator\* getStdAllocator();

**Comment by vrabaud** [comment]
ok. You are proposing changes that do not compile under OpenCV 2.4.*. Please include a pre-processor #if to compare the OpenCV versions and use the right code, thx.

**Comment by edgarriba** [comment]
the changes of this repo are not mine. I have a fork with the versions comparissons but i'm not sure how to pull request from a fork of a fork.

https://github.com/edgarriba/vision_opencv

**Comment by vrabaud** [comment]
should be the same really. But your fork also need to deal with the refcount and g_numpyAllocator ?

**Comment by vrabaud** [comment]
hey, I am at ROScon right now. I released OpenCV3 as a deb and I am planning to test your fixes with it !

**Comment by edgarriba** [comment]
nice!

**Pull request closed without merging**